features/FEATURE_LWIP/lwip-interface/lwip-sys/arch/lwip_sys_arch.c

```diff
@@ -123,15 +123,15 @@ u32_t sys_now(void) {
  *---------------------------------------------------------------------------*/
 err_t sys_mbox_new(sys_mbox_t *mbox, int queue_sz) {
     if (queue_sz > MB_SIZE)
-        error("sys_mbox_new size error\n");
+        SET_ERROR(MAKE_ERROR(MODULE_NETWORK_STACK, ERROR_CODE_INVALID_SIZE), "sys_mbox_new size error\n", queue_sz);
```

Comment:
Reading this, I think we should collapse these macros. 
``` cpp
SET_ERROR(MODULE_NETWORK_STACK, ERROR_CODE_INVALID_SIZE, "sys_mbox_new size error\n", queue_sz);
// and
SET_CUSTOM_ERROR(MODULE_NETWORK_STACK, ERROR_CODE_INVALID_SIZE, "sys_mbox_new size error\n", queue_sz);
```

Is the extra macro name needed?

Reply:
Thats good idea, if we collapse do you any suggestions for using the custom defined error?
If we use the same macro, how does the caller use a custom defined error?

Reply:
By the way, error logging can be completely disabled using a config mechanism, but by default it is enabled.

Reply:
Does having two macros, SET_ERROR and SET_CUSTOM_ERROR not work?